src/inc/quic_storage.h

```diff
@@ -69,6 +78,33 @@ CxPlatStorageReadValue(
     _Inout_ uint32_t * BufferLength
     );
 
+typedef enum CXPLAT_STORAGE_TYPE {
+    CXPLAT_STORAGE_TYPE_INTEGER32,
+    CXPLAT_STORAGE_TYPE_INTEGER64,
+    CXPLAT_STORAGE_TYPE_BINARY
+} CXPLAT_STORAGE_TYPE;
+
+//
+// Write a value to the storage context
+//
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+CxPlatStorageWriteValue(
+    _In_ CXPLAT_STORAGE* Storage,
+    _In_z_ const char * Name,
+    _In_ CXPLAT_STORAGE_TYPE Type,
```

Comment:
nit, for the future:
It might be interesting to pass the type to read too - if only to ensure the value read is of the expected type.

Reply:
How could this be used to help admins/users not make mistakes when creating the reg keys?  Log an error when the registry isn't the expected type and then fail the read?